News/Controllers/UsersController.cs

```diff
@@ -19,25 +20,50 @@ public UsersController(IUserService service, ITokenService tokenService)
     }
 
     [HttpGet("{id}")]
-    public async Task<IActionResult> GetUserById([FromRoute] Guid id)
+    public async Task<IActionResult> GetAsync([FromRoute] Guid id, CancellationToken cancellationToken)
     {
-        var user = await _service.GetUserById(id);
+        var user = await _service.GetAsync(id, cancellationToken);
+        if (user is null)
+        {
+            return NotFound($"User with ID {id} was not found.");
+        }
 
         return Ok(user);
     }
 
     [HttpPatch("{id}")]
-    public async Task<IActionResult> UpdateUser([FromRoute] string id, [FromBody] JsonPatchDocument<UserUpdateModel> user)
+    public async Task<IActionResult> UpdateAsync(
+        [FromRoute] string id, 
+        [FromBody] JsonPatchDocument<UserUpdateModel> user,
+        CancellationToken cancellationToken
+        )
     {
-        var updatedUser = await _service.UpdateUser(user, id);
+        if (user is null)
+        {
+            return BadRequest("Patch document cannot be null.");
+        }
+
+        var existingUser = await _service.GetAsync(Guid.Parse(id), cancellationToken);
+        if (existingUser is null)
+        {
+            return NotFound($"User with ID {id} was not found.");
+        }
+
+        var updatedUser = await _service.UpdateAsync(user, id, cancellationToken);
 
         return Ok(updatedUser);
     }
 
     [HttpDelete]
-    public async Task<IActionResult> DeleteUser([FromQuery] Guid id)
+    public async Task<IActionResult> DeleteAsync([FromQuery] Guid id, CancellationToken cancellationToken)
     {
-        await _service.DeleteUser(id);
+        var existingUser = await _service.GetAsync(id, cancellationToken);
+        if (existingUser is null)
+        {
+            return NotFound($"User with ID {id} was not found.");
+        }
```

Comment:
Is it good practice to find an object before do some logic? I mean can we get this error if user doesn't exist directly from db?